### ISSUE TITLE
In theory, does this project support all processors, such as PPC?

### ISSUE THREAD

**Issue by snikeguo** [CosmosOS/XSharp#3]
such as  MPC56XX(automotive electronics Chip)
ARM：STM32
and so on .

**Comment by jp2masa** [comment]
At the moment, X# being reworked and there are plans for ARMv7 support. It shouldn't be hard to add support for other architectures, as X# will be easy to extend when the rework is finished.

**Comment by snikeguo** [comment]
Different architectures have different register definitions, which reminds me of the C language header file and the C # namespace. . aha!
Anyway, I am looking forward to this project is completed, and then I will be the project to support the processor extended to the MCU (arm cortex-M0 / 3/4/7, PPC E200Z0) 
When I needed to port an RTOS (such as Free RTOS) to a new architecture CPU, I had to look at the CPU's data sheet. This is very upset, all the embedded engineers are like this. If this project is completed, then RTOS port to the new CPU will be very simple.
This is a very good project！
